### PR TITLE
Fixed Y-axis and y2-axis icons not updating during reflow

### LINKED ISSUE #87: Y-axis icons and y2-axis icons not updating during reflow
# Bug Report

## Description

Y-axis icons and y2-axis icons not updating based on the values passed down to the reflow. If there is no data for a particular legend, the y-axis icon should hide and if there is data for a particular legend during panning, y-axis icons should display.

## Link to CodeSandbox
Use [this](https://codesandbox.io/s/ecstatic-heisenberg-5tugf?file=/src/data.js) template to reproduce the behavior.

## Expected Behavior

If there is no data for a particular legend during panning, the y-axis icons should hide and if there is data for a particular legend during panning, the y-axis icon should display. Same should happen for y2 as well.

### PR DESCRIPTION
**Pull request by AshishMotanamGurunadham**
### Summary

Y-axis icons and y2-axis icons not updating based on the values passed down to the reflow. If there is no data for a particular legend, the y-axis icon should hide and if there is data for a particular legend during panning, y-axis icons should display.

Closes #87 

### Deployment Link

https://terra-graphs-deployed-pr-#.herokuapp.com/

#### Before(during reflow)
image

#### After(during reflow)
image

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

File: packages/carbon-graphs/src/js/controls/Bar/Bar.js [hunk 1]
```diff
@@ -258,6 +258,22 @@ class Bar extends GraphContent {
         this.dataTarget,
         graph.config,
       );
+    } else if (graph.config.axis.y2.show) {
+      /* If data belongs to yAxis and shape associated with data id in yAxis is null then prepareLabelShapeItem method gets executed to add shape in yAxis
+       else if data belongs to y2Axis and shape associated with data id in y2Axis is null then prepareLabelShapeItem method gets executed to add shapes in y2Axes. */
+      if (this.config.yAxis === 'y' && (graph.svg.select(`.${styles.axisLabelYShapeContainer} svg[aria-describedby="${graphData.key}"]`).empty())) {
```
Comment (yuderekyu) [inline packages/carbon-graphs/src/js/controls/Bar/Bar.js:264]: can we save this query into a variable, then use it within the inner two conditionals? i'm not sure what to call it
```suggestion
      const yAxisContainer = graph.svg.select(`.${styles.axisLabelYShapeContainer} svg[aria-describedby="${graphData.key}"]`)
```

File: packages/carbon-graphs/src/js/controls/Line/Line.js [hunk 2]
```diff
@@ -256,6 +256,22 @@ class Line extends GraphContent {
         this.dataTarget,
         graph.config,
       );
+    } else if (graph.config.axis.y2.show) {
```
Comment (yuderekyu) [inline packages/carbon-graphs/src/js/controls/Line/Line.js:259]: is this the same exact logic as the bar? if so, is it possible to extract this into pure function for reuse?

File: packages/carbon-graphs/src/js/controls/Bar/Bar.js (hunk 1 above)
Comment (AshishMotanamGurunadham) [reply packages/carbon-graphs/src/js/controls/Bar/Bar.js:264]: This is one of the rule points we have, to not assign things to variable if it is only used in one place. If it is different for terra I can change it.

Comment (yuderekyu) [reply packages/carbon-graphs/src/js/controls/Bar/Bar.js:264]: oh, the style identifier is different. that tells me we could abstract this into a pure function that accepts a container param. take this with a grain of salt though, I'm still not super familiar with the project:
```js
const isYAxisContainerEmpty = (container) => graph.svg.select(`.${container} svg[aria-describedby="${graphData,key}"]`).empty();

if (this.config.yAxis === 'y' && isYAxisContainerEmpty(styles.axisLabelYShapeContainer)) { }
if (this.config.yAxis === 'y' && isYAxisContainerEmpty(styles.axisLabelY2ShapeContainer)) { }
```

**Review by sdadn — APPROVED** [review]

File: packages/carbon-graphs/src/js/controls/Line/Line.js (hunk 2 above)
Comment (AshishMotanamGurunadham) [reply packages/carbon-graphs/src/js/controls/Line/Line.js:260]: Fixed [here](https://github.com/cerner/terra-graphs/pull/94/commits/db69bb1aa18d032ad4a1a98365fe5dc2bb8316cc#diff-a75e9754db9269401928d444c9f036bc068af538ee5ca551e761767a5394e23eR339).

File: packages/carbon-graphs/src/js/helpers/label.js [hunk 3]
```diff
@@ -336,6 +336,29 @@ const destroyTooltipDiv = () => {
   }
 };
 
+/**
+ * Update shapes in y and y2 axes during reflow.
+ *
+ * @private
+ * @param {object} graph - Graph instance
+ * @param {object} graphData - Graph data
+ * @param {Line} control - Line instance
```
Comment (mraplumb) [inline packages/carbon-graphs/src/js/helpers/label.js:345]: nit:  is this really a Line?  Looks to be central to multiple graphs.  Should this be updated?

Comment (AshishMotanamGurunadham) [reply packages/carbon-graphs/src/js/helpers/label.js]: Fixed [here](https://github.com/cerner/terra-graphs/pull/94/commits/98f1f881fd37737792f72504fb242b88800e2f7d#diff-a75e9754db9269401928d444c9f036bc068af538ee5ca551e761767a5394e23eR345).

**Review by ryanthemanuel — APPROVED** [review]

**Review by mraplumb — APPROVED** [review]

**Review by dkasper-was-taken — APPROVED** [review]

**Pull request merged**